### ISSUE TITLE
[Request] Folders monitoring

### ISSUE THREAD

**Issue by mf35** [fupdec/Adult-Video-Database#14]
We alluded to it so i put it in a Github issue so you can track it.
Always with the aim to get things better and more user friendly, i proposed to upgrade the way you can add new videos in the library.
It can echoes with #12 maybe.

For now, each time you want to add new videos, you have to:
- Click the "Add new videos" button
- Click the "Choose folders" button
- Seek the good foler on your computer
- Click continue
- Choose if you want to scan the name of the files for tags
- Start the scan
- Wait for the scan
- Click "Ok"

Maybe it could be less tedious if you had an "Add new video folder" button with an "Update library" one.
So imagine the user have videos in 3 folders with sub-folders.
- Click the "Add new video folder" button
- Seek and select the 3 folders just for once
- Click continue
- Choose if you want to scan the name of the files for tags
- Start the scan
- Wait for the scan
- Click "Ok"

Now the user have a list with your 3 monitored folder. You can delete them (from the list), change their path, ...
When the user adds or deletes videos from its computer monitored folder, it could be mirrored in the app on demand:
- Click the "Update library" button
- Choose if you want to scan the name of the files for tags
- Start the scan
- Wait for the scan
- Click "Ok"

Here AVDB would have just checked for new videos and added them to the library, and checked for deleted or renamed videos and asked the user what to do with them.
Maybe adding an option to scan sub-folders or not could be useful to some people.

I think this process should be manual, but it can still dynamically alert the user (at particular moments or on particular events) that a monitored folder content has changed. With a highlighted button with the number of new videos or something.

**Comment by fupdec** [comment]
> I think this process should be manual, but it can still dynamically alert the user (at particular moments or on particular events) that a monitored folder content has changed. With a highlighted button with the number of new videos or something.

I checked out the "chokidar" module, it is great for keeping track of new files in a folder. I've already configured it to track new files with the extension of all known video files. works great. it monitors files in real time without overloading the main application process. that is, it will run in the background while the application is running.

but there are points to consider. for example, how to save a list of all tracked files that have already been added to the database and how to initialize it later when the application starts.

that is, there is no problem keeping track of new files when the application is running. But after all, we need to find out if new files have appeared when the application starts. in general I think it is realizable. I'm just informing you that this task is in progress.
